packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts

```diff
@@ -416,6 +423,46 @@ export class IndyVdrAnonCredsRegistry implements AnonCredsRegistry {
       revocationRegistryDefinitionMetadata: {},
     }
   }
+
+  public async fetchIndySchemaWithSeqNo(agentContext: AgentContext, seqNo: number, did: string, type: number) {
+    const indyVdrPoolService = agentContext.dependencyManager.resolve(IndyVdrPoolService)
+
+    const pool = await indyVdrPoolService.getPoolForDid(agentContext, did)
+
+    agentContext.config.logger.debug(`Getting transaction with seqNo '${seqNo}' from ledger '${pool.indyNamespace}'`)
+    const request = new GetTransactionRequest({ ledgerType: type, seqNo })
+
+    agentContext.config.logger.trace(`Submitting get transaction request to ledger '${pool.indyNamespace}'`)
+    const response = await pool.submitReadRequest(request)
+
+    if (response.result.data?.txn.type === '101') {
+      const schema = response.result.data?.txn.data as SchemaType
+
+      const schemaId = getLegacySchemaId(did, schema.data.name, schema.data.version)
+
+      return {
+        schema: {
+          schemaId,
+          attr_name: schema.data.attr_names,
+          name: schema.data.name,
+          version: schema.data.version,
+          issuerId: did,
+          seqNo,
+        },
+        indyNamespace: pool.indyNamespace,
+      }
+    }
+
+    return agentContext.config.logger.error(`Could not get schema from ledger for seq no ${seqNo}'`)
```

Comment:
In general I always write the unhappy flow first, and return early in that case. This way you can often prevent big if statement blocks, and simplify the code. This article explain the principle quite nicely I think: https://medium.com/swlh/return-early-pattern-3d18a41bba8

So in this case we'd first check if the type is _not_ equal to 101, and if that's the case we return null. Then the main happy flow of this function (the biggest code chunk) will return on the top level of the method.

Reply:
Also we return a logger call here, I think that should be null:

```suggestion
    agentContext.config.logger.error(`Could not get schema from ledger for seq no ${seqNo}'`)
    return null
```